### PR TITLE
Feature/search from query object

### PR DESCRIPTION
**Pull request by danielronnkvist**
- Find tags from search object
- Find files from filetypes in search object
- Find files matching a fuzzy search string
- Find where the resulting arrays are matching
- Return the matches and make sure that the result is uniq
- Make sure current_user owns the returning files

Also added some tests to make sure it works.

### CODE REVIEW

**Comment by ronjagrosz** [comment]
This has probably something to do with the code in develop, because I get this in my testing as well, should this be ignored:
_Error:
SearchTest#test_visits_the_search_page:
Cliver::Dependency::NotFound: Could not find an executable ["phantomjs"] on your path._

**Comment by danielronnkvist** [comment]
@Fluffpannkaka Does it help if you run `sudo apt-get install phantomjs`? Phantomj is a headless javascript engine that we use to compile the javascript when testing. You can read more [here](http://phantomjs.org/) if you like.

**Comment by ronjagrosz** [comment]
Of course, should have thought of that. Works perfectly now! I'll continue reading through the pull request now. :dancer:

**Comment by danielronnkvist** [comment]
Sweeet :bee:

**Comment by ronjagrosz** [comment]
Looks good! :baby_chick:

File: config/routes.rb [hunk 1]
```diff
@@ -26,6 +26,6 @@
   get '/get/:download_hash/:name', to: 'files#download', :via => :all
 
   # search
-  get '/search/:query', to: 'search#find', :defaults => { :format => 'json' }
+  post '/search/api', to: 'search#find', :defaults => { :format => 'json' }
```
Comment (klaseskilson) [inline config/routes.rb]: is the `/api` part of the route necessary?

**Pull request merged**